### ISSUE TITLE
Support drag and drop for re-ordering space rooms and sub-spaces

### ISSUE THREAD

**Issue by t3chguy** [vector-im/element-web#16992]
Blocked on the removal of Groups which holds a dependency on a very old version of react-beautiful-dnd which is non-trivial to replace so removal is the only option.

**Comment by robintown** [comment]
Depending on two different versions of react-beautiful-dnd is always an option, right? Apart from bundle size concerns

**Comment by t3chguy** [comment]
Whilst yes webpack and npm support dependency aliasing for this purpose, there is 0 guarantee from the lib that it supports this.

It might register globals which conflict, I think the time wasted debugging issues with a dnd library is better spent on other things and dnd can come later.

**Comment by daniellekirkwood** [comment]
I'm going to remove the Needs-Product label as it would be great to have a drag&drop functionality in the app. 

Ideally; 
* Subspaces can be drag&drop to move the ordering in the left panel (similarly to spaces). 
- [ ] Question: Can subspaces be drag&drop to different Spaces? Should we do that Design, can we do that Engineering?
* Rooms can be drag&drop into Spaces or subspaces.

**Comment by ShadowJonathan** [comment]
Additional questions;
- [ ] If you drag a subspace between two spaces you own, would they "copy", or "move"?
- [ ] Similarly, if you drag a subspace from a non-owned space to an owned one, would they unambiguously copy?
- [ ] What about dragging a subspace from a space to the main bar? Would it be "pinned" in the sidebar? (Would this be "move" or "copy" as well?)

**Comment by t3chguy** [comment]
Keep in mind sometimes permissions might make it so you cannot `move` as you may not have the required permission to remove it from the "from" space.

> What about dragging a subspace from a space to the main bar? Would it be "pinned" in the sidebar? (Would this be "move" or "copy" as well?)

This is out of scope as it'd require spec work.

**Comment by noantiq** [comment]
Hi! after opening an issue for dragging rooms into spaces (https://github.com/vector-im/element-web/issues/22105) I was pointed here. I missed that a comment in this issue already reflects the use case since the issue title led me believe that this is only about re-ordering :D Maybe it would help to change the title or description of this issue accordingly?

**Comment by Reinachan** [comment]
Just adding that there is a spec for space maintainers to decide the room order in Matrix, `m.space.child -> content -> order` I'd suggest using that for specifying the order rooms appear in.
https://spec.matrix.org/v1.3/client-server-api/#ordering-1
